syntax = "proto3";

package analysis;

service AnalysisService {
  // Analyzes a video and returns a stream of progress updates and final results.
  rpc AnalyzeVideo(AnalysisRequest) returns (stream AnalysisResponse);
}

// Request to analyze a video file
message AnalysisRequest {
  string video_path = 1;           // Path to input video file
  string match_id = 2;             // Unique match identifier
  int32 frame_limit = 3;           // Max frames to process (0 = all)
  bool skip_json = 4;              // Skip JSON export if true
  float confidence_threshold = 5;  // YOLO confidence threshold
}

// Streaming response with progress and final results
message AnalysisResponse {
  string match_id = 1;
  string status = 2;               // PENDING, PROCESSING, COMPLETED, FAILED
  float progress = 3;              // 0.0 to 1.0
  string message = 4;
  
  // Final results (only populated when status is COMPLETED)
  AnalysisResult result = 5;
}

// Complete analysis results
message AnalysisResult {
  string match_id = 1;
  int32 total_frames = 2;
  int32 players_tracked = 3;
  
  // Output file paths (relative to outputs directory)
  string tracking_video_path = 4;           // tracking_full_output.mp4
  string tracking_json_path = 5;            // tracking_results.json
  string backline_video_path = 6;           // backline_analysis.mp4
  string heatmap_video_path = 7;            // heatmap_analysis.mp4
  string possession_analysis_path = 8;      // possession_analysis_results.json
  string animation_video_path = 9;          // tracking_animation.mp4
  
  // Performance metrics
  double processing_time_seconds = 10;
  double fps = 11;
  
  // Error information (if any)
  string error_message = 12;
}
